src/app/ApplicationTemplate.Presentation/Framework/Analytics/AnalytcsSink.cs

```diff
@@ -0,0 +1,146 @@
+		if (navigatorState.LastRequestState != NavigatorRequestState.Processed)
+		{
+			// Skip the requests that are still processing of that failed to process.
+			return;
+		}
+
+		// Get the actual ViewModel instance.
+		// This allows to track based on instances and not types (because there are scenarios where you can open the same page multiple times with different parameters).
+		// Having the instance also allows casting into more specific types to get more information, such as navigation parameters, that could be relevant for analytics.
+		var viewModel = navigatorState.GetActiveStackNavigator().State.Stack.LastOrDefault()?.ViewModel;
+		if (viewModel is null || _lastViewModel == viewModel)
+		{
+			return;
+		}
+
+		// Gather analytics data.
+		var pageName = viewModel.GetType().Name.Replace("ViewModel", string.Empty, StringComparison.OrdinalIgnoreCase);
+		var isInModal = navigatorState.ActiveModal != null;
+		var sectionName = navigatorState.ActiveSection.Name;
+
+		// Send the analytics event.
+		SendPageView(pageName, isInModal, sectionName);
+
+		// Capture the last ViewModel instance to avoid duplicate events in the future.
+		_lastViewModel = viewModel;
+	}
+
+	private void SendPageView(string pageName, bool isInModal, string sectionName)
+	{
+		// TODO: Implement page views using a real analytics provider.
+		if (!_logger.IsEnabled(LogLevel.Information))
+		{
+			return;
+		}
+
+		if (isInModal)
+		{
+			_logger.LogInformation("Viewed page '{PageName}' in modal.", pageName);
+		}
+		else
+		{
+			_logger.LogInformation("Viewed page '{PageName}' in section '{SectionName}'.", pageName, sectionName);
+		}
+	}
+
+	public void TrackDataLoaderSuccess(IDataLoaderRequest request, object value)
+	{
+		// TODO: Implement data loaded events using a real analytics provider.
+		if (!_logger.IsEnabled(LogLevel.Information))
+		{
+			return;
+		}
+
+		var dataLoaderName = request.Context.GetDataLoaderName();
+		if (request.Context.TryGetValue("IsForceRefreshing", out var isForceRefreshing) && isForceRefreshing is true)
+		{
+			_logger.LogInformation("Refreshed content '{DataLoaderName}'.", dataLoaderName);
+		}
+		else
+		{
+			_logger.LogInformation("Loaded content '{DataLoaderName}'.", dataLoaderName);
+		}
+	}
+
+	public void TrackDataLoaderFailure(IDataLoaderRequest request, Exception exception)
+	{
+		// TODO: Implement data loaded events using a real analytics provider.
+		if (!_logger.IsEnabled(LogLevel.Information))
+		{
+			return;
+		}
+
+		var dataLoaderName = request.Context.GetDataLoaderName();
+		if (request.Context.TryGetValue("IsForceRefreshing", out var isForceRefreshing) && isForceRefreshing is true)
+		{
+			_logger.LogInformation("Failed to refresh content '{DataLoaderName}'.", dataLoaderName);
+		}
+		else
+		{
+			_logger.LogInformation("Failed to load content '{DataLoaderName}'.", dataLoaderName);
+		}
+	}
+
+	public void TrackCommandSuccess(string commandName, object? commandParameter, WeakReference<IViewModel>? viewModel)
```

Comment:
just to make sure. Is the goal to use the ServiceProvider from each vm to get the analytics service to properly track every event?